src/main/java/org/opensearch/knn/index/codec/util/KNNCodecUtil.java

```diff
@@ -112,7 +113,17 @@ public static String buildEngineFilePrefix(String segmentName) {
     }
 
     public static String buildEngineFileSuffix(String fieldName, String extension) {
-        return String.format("_%s%s", fieldName, extension);
+        return String.format("_%s%s", escapeSpaceInFieldName(fieldName), extension);
+    }
+
+    private static String escapeSpaceInFieldName(String fieldName) {
```

Comment:
```suggestion
    private static String escapeSpaceInFieldName(final String fieldName) {
```

Reply:
Sounds good! Will update in the next revision.